### DISCUSSION TITLE
Dimenet forward pass tensor size mismatch

### DISCUSSION THREAD

**Discussion by sudaksh14** [pyg-team/pytorch_geometric#6217] [Q&A]
Hi Guys,
I am using dimenet for my project since 2 months now and its giving great results. Thanks for all your efforts in putting this library together.
I am facing an issue with a particular molecule type which has 12 atoms nd hence pos matric has a shape of (12,3) and the z parameter has shape (12). When I am passing these through the forward function I am getting this error:
![image](https://user-images.githubusercontent.com/102596407/207036914-c9c66a50-25ee-4022-91cb-83cd87c3b711.png)

This happens with both dimenet and dimenetPP, I could not imagine why its not the case when pos shape is small say (3,3).

Thanks in advance!

**Comment by rusty1s** [comment]
Can you share your molecule with me?

**Reply by sudaksh14** [reply]
sure its benzene with the following data in csv, I have a preprocessing to convert them into matrices of Batchesx12x3(pos) and then I consider the z param as [6,6,6,6,6,6,1,1,1,1,1,1] as per the atomic nos.
[ammonia_data.csv](https://github.com/pyg-team/pytorch_geometric/files/10237218/ammonia_data.csv)

**Reply by rusty1s** [reply]
Thank you! Is it possible for you to just share the PyG data/batch for which your model fails, e.g., by running `torch.save(batch, 'batch.pt')`?

**Reply by sudaksh14** [reply]
Ok I understand now.
Here is the saved train batch, I have zipped it as .pt files can't be uploaded directly
[train_batch.zip](https://github.com/pyg-team/pytorch_geometric/files/10305974/train_batch.zip)

**Reply by rusty1s** [reply]
Thank you! I fixed it in https://github.com/pyg-team/pytorch_geometric/pull/6305.

**Reply by sudaksh14** [reply]
Hi matthias,
Its still gives me error but the shapes are matching now. Can you suggest what to do?

![image](https://user-images.githubusercontent.com/102596407/212062306-d87110c0-b53a-4561-bd8f-f77a130cb020.png)

**Reply by rusty1s** [reply]
Removing line 602 should fix this. Looks like you duplicated it.

**Reply by sudaksh14** [reply]
yes, I did that mistake only but it works fine now
Thnx for the help
